app/api/domains/osu.py

```diff
@@ -1951,14 +1951,9 @@ async def register_account(
             ip = app.state.services.ip_resolver.get_ip(request.headers)
 
             if not ip.is_private:
-                if app.state.services.geoloc_db is not None:
-                    # decent case, dev has downloaded a geoloc db from
-                    # maxmind, so we can do a local db lookup. (~1-5ms)
-                    # https://www.maxmind.com/en/home
-                    geoloc = app.state.services.fetch_geoloc_db(ip)
-                else:
-                    # worst case, we must do an external db lookup
-                    # using a public api. (depends, `ping ip-api.com`)
+                geoloc = app.state.services.fetch_geoloc_nginx(ip, request.headers)
+
+                if geoloc is None:
                     geoloc = await app.state.services.fetch_geoloc_web(ip)
 
                 if geoloc is not None:
```

Comment:
would this ever be reached even? localhost + no db or what? i would like some more testing on this PR

```py
if geoloc is not None:
```

also i think while we are refactoring this we could have this a method that parses headers

1. cloudflare first
2. nginx headers second
3. XX as fallback

and move logic outside from here as it could potentially be used elsewhere as well

Reply:
it can be reached if web api is in use & is down

Reply:
hmm i think i like the 1/2/3 idea